mousebender/simple.py

```diff
@@ -263,8 +263,27 @@ def handle_starttag(
         # PEP 658:
         # ... each anchor tag pointing to a distribution MAY have a
         # data-dist-info-metadata attribute.
-        if "data-dist-info-metadata" in attrs:
-            found_metadata = attrs.get("data-dist-info-metadata")
+        # Addendum: PEP 714
+        #   The PEP 658 metadata, when used in the HTML representation of the Simple
+        #   API, MUST be emitted using the attribute name data-core-metadata, with
+        #   the supported values remaining the same.
+        found_metadata = ""
+        if "data-core-metadata" in attrs:
+            found_metadata = attrs.get("data-core-metadata")
+
+            # Addendum: PEP 714
+            #   ...To support clients that used the previous key names, the HTML
+            #   representation MAY also be emitted using the data-dist-info-metadata, and
+            #   if it does so it MUST match the value of data-core-metadata.
+            if ("data-dist-info-metadata" in attrs) and (
+                found_metadata != attrs.get("data-dist-info-metadata")
+            ):
+                # Because of the use of MUST we will throw an exception in this case.
+                raise ValueError(
+                    "PEP 714 Violation. `data-dist-info-metadata` is present in package ",
+                    "metadata, but it does not match the value given in `data-core-metadata`",
+                )
```

Comment:
```suggestion
                raise ValueError(
                    "'data-dist-info-metadata' differs from 'data-core-metadata'",
                )
```

We should probably raise a new `InvalidDistroMetadata` exception.